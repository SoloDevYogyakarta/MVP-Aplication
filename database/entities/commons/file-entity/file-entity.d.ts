import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class FileEntity extends BaseEntity {
    filename: string;
    originalname: string;
    filepath: string;
    type: string;
    width: number;
    height: number;
}
export type FileCreationAttribute = Optional<FileEntity, 'id'>;
export interface FileInstance extends Model<FileCreationAttribute, FileEntity>, FileEntity {
}
export declare const fileEntity: import("sequelize").ModelCtor<FileInstance>;
