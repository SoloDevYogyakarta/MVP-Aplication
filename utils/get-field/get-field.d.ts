import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
