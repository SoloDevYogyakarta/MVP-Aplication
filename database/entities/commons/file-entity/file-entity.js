"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fileEntity = exports.FileEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class FileEntity extends base_entity_1.BaseEntity {
}
exports.FileEntity = FileEntity;
exports.fileEntity = entity_1.sequelize.define('FILE', {
    public_id: sequelize_1.DataTypes.STRING,
    filename: sequelize_1.DataTypes.STRING,
    originalname: sequelize_1.DataTypes.STRING,
    filepath: sequelize_1.DataTypes.STRING,
    type: sequelize_1.DataTypes.STRING,
    width: sequelize_1.DataTypes.INTEGER,
    height: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'FILE',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=file-entity.js.map