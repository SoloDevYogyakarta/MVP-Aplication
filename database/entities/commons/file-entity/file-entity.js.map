{"version":3,"file":"file-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/commons/file-entity/file-entity.ts"],"names":[],"mappings":";;;AAAA,mCAAgC;AAChC,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,UAAW,SAAQ,wBAAU;CAOzC;AAPD,gCAOC;AAMY,QAAA,UAAU,GAAG,kBAAS,CAAC,MAAM,CACxC,MAAM,EACN;IACE,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,QAAQ,EAAE,qBAAS,CAAC,MAAM;IAC1B,YAAY,EAAE,qBAAS,CAAC,MAAM;IAC9B,QAAQ,EAAE,qBAAS,CAAC,MAAM;IAC1B,IAAI,EAAE,qBAAS,CAAC,MAAM;IACtB,KAAK,EAAE,qBAAS,CAAC,OAAO;IACxB,MAAM,EAAE,qBAAS,CAAC,OAAO;CACO,EAClC;IACE,SAAS,EAAE,MAAM;IACjB,MAAM,EAAE,SAAS;IACjB,UAAU,EAAE,KAAK;IACjB,KAAK,EAAE;QACL,YAAY,CAAC,UAAU,EAAE,OAAO;YAC9B,MAAM,QAAQ,GAAG,UAAU,CAAC;YAC5B,QAAQ,CAAC,SAAS,GAAG,IAAA,eAAM,GAAE,CAAC;QAChC,CAAC;KACF;CACF,CACF,CAAC"}