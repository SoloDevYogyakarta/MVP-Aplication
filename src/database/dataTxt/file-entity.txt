{"id":47,"public_id":"To7a9yF14XeM2kaE4PRFv","filename":null,"originalname":null,"filepath":null,"type":null,"width":0,"height":0}