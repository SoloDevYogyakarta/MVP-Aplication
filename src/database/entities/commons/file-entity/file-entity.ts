import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class FileEntity extends BaseEntity {
  filename: string;
  filepath: string;
  type: string;
}

export type FileCreationAttribute = Optional<FileEntity, 'id'>;
export interface FileInstance
  extends Model<FileCreationAttribute, FileEntity>,
    FileEntity {}
export const fileEntity = sequelize.define<FileInstance>(
  'FILE',
  {
    public_id: DataTypes.STRING,
    filename: DataTypes.STRING,
    filepath: DataTypes.STRING,
    type: DataTypes.STRING,
  } as ModelAttributes<FileInstance>,
  {
    tableName: 'FILE',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
