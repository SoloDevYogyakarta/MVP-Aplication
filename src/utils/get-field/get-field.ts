import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
