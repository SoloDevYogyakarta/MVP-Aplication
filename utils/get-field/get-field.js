"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map