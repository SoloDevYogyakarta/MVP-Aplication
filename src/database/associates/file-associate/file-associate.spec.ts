import { ModelCtor } from 'sequelize';
import { FileInstance } from '../../../database/entities/commons/file-entity/file-entity';
import { getField } from '../../../utils/get-field/get-field';
import {
  fileAttribute,
  fileUserAssociate,
  fileUserInclude,
} from './file-associate';

describe('fileUserAssociate', () => {
  let public_id!: string;
  let entity: ModelCtor<FileInstance>;

  beforeEach(() => {
    entity = fileUserAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    public_id = getField('file-entity').public_id;
  } catch (err) {
    // empty
  }

  if (public_id) {
    it('findOne with relationship', async () => {
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll({
        attributes: fileAttribute,
        include: fileUserInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  }
});
