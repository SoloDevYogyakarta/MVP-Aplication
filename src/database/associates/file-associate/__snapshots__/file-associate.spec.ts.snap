// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileUserAssociate render correctly 1`] = `[Function]`;
